extensions/mermaid-to-image/src/mermaid-to-image.tsx

```diff
@@ -0,0 +1,184 @@
+import React, { useEffect, useState, useRef } from "react";
+import { showToast, Toast, Detail, Icon, ActionPanel, Action } from "@raycast/api";
+import { showFailureToast } from "@raycast/utils";
+import { ImagePreview } from "./components/ImagePreview";
+import { generateMermaidDiagram } from "./utils/diagram";
+import { cleanupTempFile } from "./utils/files";
+import { Clipboard, getPreferenceValues } from "@raycast/api";
+import { Preferences } from "./types";
+
+export default function Command() {
+  const [imagePath, setImagePath] = useState<string | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [error, setError] = useState<string | null>(null);
+  const isProcessingRef = useRef(false);
+  const tempFileRef = useRef<string | null>(null);
+  const preferences = getPreferenceValues<Preferences>();
+
+  async function processMermaidCode() {
+    if (isProcessingRef.current) return;
+    isProcessingRef.current = true;
+
+    try {
+      setIsLoading(true);
+      setError(null);
+
+      // Use clipboard content
+      let mermaidCode;
+      try {
+        mermaidCode = await Clipboard.readText();
+        if (!mermaidCode) {
+          setError("Clipboard is empty. Please copy a Mermaid diagram code first.");
+          await showFailureToast({
+            title: "Empty Clipboard",
+            message: "Please copy a Mermaid diagram code first.",
+          });
+          setIsLoading(false);
+          return;
+        }
+      } catch (error: unknown) {
+        const errorMessage = error instanceof Error ? error.message : String(error);
+        setError("Failed to read clipboard. Please try again.");
+        await showFailureToast({
+          title: "Clipboard Error",
+          message: "Failed to read clipboard. Please try again.",
+        });
+        console.error("Clipboard error details:", errorMessage);
+        setIsLoading(false);
+        return;
+      }
+
+      await showToast({
+        style: Toast.Style.Animated,
+        title: "Generating diagram...",
+      });
+
+      const outputPath = await generateMermaidDiagram(mermaidCode, tempFileRef);
+      setImagePath(outputPath);
+
+      await showToast({
+        style: Toast.Style.Success,
+        title: "Diagram generated successfully",
+      });
+    } catch (error) {
+      // Log the full error for debugging but show a simplified version to the user
+      console.error("Error details:", error);
+
+      // Extract a user-friendly message
+      let userMessage = "An unexpected error occurred.";
+      if (error instanceof Error) {
+        userMessage = error.message;
+      } else if (typeof error === "string") {
+        userMessage = error;
+      }
+
+      setError(userMessage);
+
+      // Show failure toast with the user-friendly message
+      await showFailureToast({
+        title: "Diagram Generation Failed",
+        message: userMessage,
+      });
```

Comment:
If you're inside a catch you can do like this
```suggestion
      await showFailureToast(error, {
        title: "Diagram Generation Failed",
        message: userMessage,
      });
```

if not inside a catch it's better to use the regular toast (since this uses error)

Reply:
@pernielsentikaer 
Thank you for your valuable suggestions 😊